### PR TITLE
Recipe for openbabel

### PR DESCRIPTION
**Pull request by njzjz**
Copied from [openbabel/conda-openbabel](openbabel/conda-openbabel) and submitted per openbabel/conda-openbabel#2.

This closes openbabel/conda-openbabel#2.

### CODE REVIEW

**Comment by njzjz** [comment]
> Hi! This is the friendly automated conda-forge-linting service.
> 
> I just wanted to let you know that I linted all conda-recipes in your PR (`recipes/openbabel`) and found it was in an excellent condition.
> 
> I do have some suggestions for making it better though...
> 
> For **recipes/openbabel**:
> 
> * Recipe with the same name exists in bioconda: please discuss with @conda-forge/bioconda-recipes.

What should I do? @conda-forge/bioconda-recipes

**Comment by njzjz** [comment]
The error seems to be fixed by openbabel/openbabel#1564.

File: recipes/openbabel/build.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      -DPYTHON_LIBRARY=$PYTHON_LIBRARY \
+      -DPYTHON_EXECUTABLE=$PYTHON \
+      -DPYTHON_INCLUDE_DIR=$PYTHON_INCLUDE_DIR \
+      -DPYTHON_BINDINGS=ON \
+      -DLIBXML2_INCLUDE_DIR=$PREFIX/include \
+      -DLIBXML2_LIBRARIES=$PREFIX/lib/libxml2.${SO_EXT} \
+      -DZLIB_INCLUDE_DIR=$PREFIX/include \
+      -DZLIB_LIBRARY=$PREFIX/lib/libz.${SO_EXT} \
+      -DRUN_SWIG=ON \
+      -DCMAKE_SHARED_LINKER_FLAGS='-pthread -lrt -lutil'
```
Comment (njzjz) [inline recipes/openbabel/build.sh]:
```suggestion
      -DCMAKE_SHARED_LINKER_FLAGS_INIT='-pthread -lrt -lutil'
```

**Comment by njzjz** [comment]
@conda-forge/help-c-cpp @conda-forge/help-python-c  please review.
AppVeyor failed because of no CMakeLists.txt found in the Python 3.7 environment, but it worked well in Python 2.7 and 3.6. I think it maybe an issue of AppVeyor. Is there anyone helping me to restart the job?

File: recipes/openbabel/bld.bat [hunk 2]
```diff
@@ -0,0 +1,18 @@
+cmake ^
+      -G "%CMAKE_GENERATOR%" ^
+      -DCMAKE_INSTALL_PREFIX=%PREFIX% ^
```
Comment (mcs07) [inline recipes/openbabel/bld.bat]: I wonder if this line is changed to `-DCMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%" ^` then at least the `xcopy %PREFIX%\bin %PREFIX%\Library\bin /e /c` should not be needed?

Adding `-DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX% ^` may also help.

File: recipes/openbabel/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+build:
+  number: 0
+  detect_binary_files_with_prefix: true
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
```
Comment (mcs07) [inline recipes/openbabel/meta.yaml]: Maybe add `- make  # [unix]` to the build requirements.

File: recipes/openbabel/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  host:
+    - python
+    - zlib
+    - swig
```
Comment (mcs07) [inline recipes/openbabel/meta.yaml]: I think `swig` is a build requirement not host.

File: recipes/openbabel/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      Open Babel is a chemical toolbox designed to speak the many languages of chemical
+      data. It's an open, collaborative project allowing anyone to search, convert,
+      analyze, or store data from molecular modeling, chemistry, solid-state materials,
+      biochemistry, or related areas.
+    doc_url: http://openbabel.org/
+    dev_url: https://github.com/openbabel/openbabel
+
+extra:
+  recipe-maintainers:
+    - njzjz
```
[the remark this answers is not in the document]
Comment (mwojcikowski) [reply recipes/openbabel/meta.yaml]: Sure, as we still do that for the other openbabel channel as well a lot of code might be shared

File: recipes/openbabel/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - make # [unix]
```
Comment (njzjz) [inline recipes/openbabel/meta.yaml]:
```suggestion
    - make  # [unix]
```

**Comment by njzjz** [comment]
Hi @mcs07 @mwojcikowski , AppVeyor build for Windows still failed. Do you have any idea about it? Thank you.

**Comment by beckermr** [comment]
Don't worry about appveyor

**Comment by beckermr** [comment]
Ack no do worry. I mean't azure. Sorry

File: recipes/openbabel/meta.yaml [hunk 7]
```diff
@@ -21,6 +21,7 @@ source:
 
 build:
   number: 0
+  skip: true  # [vc<14]
```
Comment (njzjz) [inline recipes/openbabel/meta.yaml]: win and py27

Comment (njzjz) [reply recipes/openbabel/meta.yaml]:
```suggestion
  skip: true  # [win and py27]
```

**Comment by chrisburr** [comment]
It looks great to me and I don't have anything to add so LGTM! 👍

**Comment by baoilleach** [comment]
FYI, Open Babel 3.0 has been tagged.

**Comment by dbast** [comment]
@baoilleach https://github.com/conda-forge/openbabel-feedstock was updated to 3.0 some days ago. Issues can be also opened directly over there.

**Comment by baoilleach** [comment]
Thanks - it seems that @njzjz is way ahead of me. I'll make a note of that repo in future.

**Pull request merged**